### ISSUE TITLE
Java.available is always false during stage=early when using gadget in script mode

### ISSUE THREAD

**Issue by cosmosgenius** [frida/frida-java#53]
I am trying to load a script onto an android app via `wrap method`. The frida lib gets injected and is loading the script as intended but `Java.available` is always false during the initial load. 

If the frida config's on_change is set to reload and once the script is reloaded (stage=late) `Java.available ` is true and everything works fine.

I needed the script to be up and running during stage=early.

script content which is compiled using frida-compile.
```
rpc.exports = {
    init: function (stage, parameters) {
        if(Java.available) {
              require('./hooks');
        }
    }
};
```

I tried doing a "Observe(setTimeout)" on `Java.available` waiting for it to be set to true before loading my hooks...but its never set to true even though the app is loaded.

**Comment by oleavr** [comment]
Hey,

This is because the lazily computed result of `Java.available` is stored even when it returns `false`. We should improve this so this is only the case when `true`. (PR welcome!)

Cheers!

**Comment by cosmosgenius** [comment]
fixed in frida/frida-java@4de996ca69f89f95d592eaf131527f60ce61ffd1
